File: EntitiesVb/CheckingAccount.vb
Public Class CheckingAccount
    Inherits Account
    Private _OverdraftAmount As String
    Public Property OverdraftAmount() As String
        Get
            Return _OverdraftAmount
        End Get
        Set(ByVal value As String)
            _OverdraftAmount = value
        End Set
    End Property
    Public Overrides Sub Withdraw(value As Decimal)
        If isBalanceAvailable(value) Then
            _balance -= value
        End If
    End Sub
    Private Function isBalanceAvailable(amount As Decimal) As Boolean
        Return Balance >= amount - OverdraftAmount
    End Function
End Class

File: EntitiesCs/CheckingAccount.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace EntitiesCs
{
    public class CheckingAccount : Account
    {
        public CheckingAccount(int number, decimal balance, decimal overdraftAmount) : base(number, balance)
        {
            OverdraftAmount = overdraftAmount;
        }
        public decimal OverdraftAmount { get; set; }    // monto de sobregiro

        public override void Withdraw(decimal value)
        {
            if (isBalanceAvailable(value)) // validación
                balance -= value; 
        }

        private bool isBalanceAvailable(decimal amount)
        {
            return Balance + OverdraftAmount >= amount;
        }
    }
}
